### ISSUE TITLE
Firewall-requirements to discover devices successfully

### ISSUE THREAD

**Issue by wasnertobias** [duncan3dc/sonos#60]
Hello,

As my $controllers are null after discovery under some network-restrictions I wonder which requirements there are to be discovered successfully.

`$sonos = new \duncan3dc\Sonos\Network;
$controllers = $sonos->getControllers();`

I especially get problems while using some routers made by FritzBox.

Can you probably give me a hint on certain ports, please?

**Comment by duncan3dc** [comment]
Discovery is done on [Port 1900](https://github.com/duncan3dc/sonos/blob/4c08609d64a9f00e5f59f7fa6c65ab76ce719b91/src/Network.php#L165)

Does the official Sonos app work on this network?

As part of version 2.0 I'm working on being able to add devices manually (#38) for cases where discovery is not an option or not desired

**Comment by wasnertobias** [comment]
Yes, everything is is working fine.

**Comment by xIntern** [comment]
I'm also curious about the requirements for discovery to work properly.
As of late my home server (win8.1) and raspberry pi (raspbian) haven't been able to discover any devices on my network. However my desktop, official app and nodejs lib have no problems.
I have been debuging this for some days now but I haven't been able to figure it out.

Any tips to help discovery?

**Comment by duncan3dc** [comment]
@xIntern That sounds strange, have you tried clearing the [cache](http://duncan3dc.github.io/sonos/setup/#cache)?

**Comment by xIntern** [comment]
Yes, however it won't create the cache since it cant find any devices. The following script discovers perfectly on my desktop, but throws the "No devices found on the current network" error on my server.

```
function var_dump_pre($var) { echo '<pre>'; var_dump($var); echo '</pre>'; }

require_once 'vendor/autoload.php';

use duncan3dc\Sonos\Network;
use duncan3dc\Sonos\Tracks\Spotify;

$today = (new DateTime)->format("Y-m-d");

$sonosDir = dirname(__FILE__);
$cacheDir = $sonosDir . '\\cache';

$cache = new Doctrine\Common\Cache\FilesystemCache($cacheDir);
$arrayCache = new Doctrine\Common\Cache\ArrayCache;

$logger = new \Monolog\Logger("sonos");
$logger->pushHandler(new \Monolog\Handler\StreamHandler($sonosDir . "\\logs\\sonos_$today.log", \Monolog\Logger::DEBUG));

$sonos = new Network($cache, $logger);

try {
    var_dump_pre($sonos->getSpeakers());
} catch (RuntimeException $e) {
    $logger->error('No devices found on the current network! Clearing cache and retrying');
    $cache->deleteAll();
    $sonos = new Network($arrayCache, $logger);
    var_dump_pre($sonos->getSpeakers());
}
```
Thanks for the quick reply

**Comment by xIntern** [comment]
I finally got it working. It seems it has something to with WinpkFilter in Windows network adapter properties. I disabled it and the lib started discovering perfectly. However, i activated it and rebooted and it still works fine so there might something weird going on.
Anyway it should be worth a try if anyone else have problems discovering devices.
[Ethernet properties](http://i.imgur.com/Vm9JZDJ.png)
